core/crypto/cbc-mac.c

```diff
@@ -77,14 +80,36 @@ static TEE_Result crypto_cbc_mac_update(struct crypto_mac_ctx *ctx,
 	}
 
 	while (len >= mc->block_len) {
+		size_t nblocks = MIN(len / mc->block_len,
+				     (size_t)CFG_CRYPTO_CBC_MAC_BUNDLE_BLOCKS);
+
+		if (nblocks > 1) {
+			out_len = nblocks * mc->block_len;
+			if (!out_tmp) {
+				out_tmp = malloc(out_len);
```

Comment:
In case `malloc()` fails we'll try this again and again for each block.
How about moving the logic to provide the temporary buffer before this loop is entered?

Reply:
Well in theory it could eventually succeed but perhaps thee chances are slim in such a tight loop. I'll fix.